File: packages/js-sdk/src/regex/detectors/ro/cnp.ts
// Romanian CNP (Cod Numeric Personal) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { roCnpChecksum } from '../../validators'

class RoCnpDetector extends RegexDetector {
  entityType = EntityType.RO_CNP
  score = 0.9
  contextKeywords = ['cnp', 'cod numeric personal', 'personal numeric code']
  contextRequired = false

  pattern = /\b[1-8]\d{12}\b/g
  validator = roCnpChecksum
}

registerRegion('ro', RoCnpDetector)

export { RoCnpDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/ro/cnp.py
"""Romanian CNP (Cod Numeric Personal) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import ro_cnp_checksum


@register_region("ro")
class CnpDetector(RegexDetector):
    entity_type = EntityType.RO_CNP
    score = 0.90
    context_keywords = [
        "cnp", "cod numeric personal", "personal numeric code",
    ]
    context_required = False
    context_window = 50

    # 13 digits starting with 1-8
    pattern = re.compile(
        r"\b[1-8]\d{12}\b"
    )
    validator = staticmethod(ro_cnp_checksum)
